src/checkers/inference/InferenceQualifierHierarchy.java

```diff
@@ -307,7 +273,7 @@ public AnnotationMirror getTopAnnotation(final AnnotationMirror am) {
             return varAnnot;
         } //else
 
-        return unqualified;
+        return inferenceMain.getRealTypeFactory().getQualifierHierarchy().getTopAnnotations().iterator().next();
```

Comment:
Ok, updated to return `this.getTopAnnotations()` which already filters out the non-VarAnnot ones in the `finish` method (line 73-91).